docs/grid/columns.md

```diff
@@ -79,6 +79,89 @@ Note that `className` and `format` support templating as described in [Templatin
+
+When `rendering.columns.distribution` is left `undefined`:
+
+- **`'full'`** is automatically selected if **no columns** specify a `width` option.
+- **`'mixed'`** is automatically selected if **at least one column** specifies a `width` option.
+
+This automatic selection simplifies configuration, as specifying widths for columns immediately enables mixed distribution.
+
+### Mixed Distribution Strategy (`'mixed'`)
+
+Use the mixed distribution when you want explicit control over individual column widths:
+
+- **Defined via**: each column’s `columns[].width` option:
+  - **Number or a string ending in `px`**: interpreted as pixels
+  - **String ending in `%`**: percentage of the table width
+- **Behavior**:
+  - Columns with defined `width` use their specified pixel or percentage value.
+  - Columns without defined `width` evenly share the remaining space, respecting each column’s `minWidth`.
+  - **Resizing**: Drag-resizing converts the widths of both the column to the left and the column to the right of the drag handle to fixed pixel values.
+
+```js
+Grid.grid('container', {
+  rendering: {
+    columns: {
+      distribution: 'mixed'
+    }
+  },
+  columns: [{
+    id: 'product',
+    width: 150 // static 150 px
+  }, {
+    id: 'price',
+    width: '20%' // 20% of table
+  }, {
+    id: 'stock'
+    // no width set, covers remaining space
+  }],
+  ...
+});
+```
+
+### Full Distribution Strategy (`'full'`)
+
+Use the full distribution when columns should proportionally fill the available table width:
+
+- **Defined via**: initial measurement of columns using CSS, stored as a ratio. Equally distributed if not defined.
+- **Behavior**:
+  - All columns proportionally share table width based on their initial CSS-measured widths.
+  - The **last column** fills remaining space (logs a warning if CSS widths exceed 100%).
+  - **Resizing**: Adjusting one column resizes both it and the adjacent column proportionally, while other columns maintain their positions and proportions.
+
+```css
+.hcg-column[data-column-id="product"] {
+    width: 50%;
+}
+```
+
+### Fixed Distribution Strategy (`'fixed'`)
+
+Use fixed distribution to keep columns at strictly defined widths:
+
+- **Defined via**: single-time measurement using CSS (defaults to 100 px).
+- **Behavior**:
+  - Columns retain their initial set pixel widths.
+  - **Resizing**: Only the dragged column changes width, causing columns to the right to be moved accordingly.
+
+```css
+.hcg-column[data-column-id="product"] {
+    width: 200px;
+}
+```
+
+### Custom Distribution Strategy
+
+If the provided distribution strategies do not meet your requirements, you can define a custom distribution strategy. You can achieve this by either modifying an existing strategy through inheritance or creating a completely new one by extending the abstract column strategy.
+
+See an example of extending the existing `'mixed'` distribution strategy [here](https://jsfiddle.net/BlackLabel/u2v4p6cd/).
```

Comment:
Demo should be moved to HC repo.

I also adjusted another demo to the basic `mixed` strategy here: https://github.com/highcharts/highcharts/pull/22968